app/src/main/java/org/schabi/newpipe/player/playback/TrackSelectionListener.java

```diff
@@ -0,0 +1,106 @@
+
+import static org.schabi.newpipe.player.Player.getResolutionStringFromFormat;
+
+import android.content.Context;
+import android.view.MenuItem;
+import android.widget.PopupMenu;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+
+import com.google.android.exoplayer2.Format;
+import com.google.android.exoplayer2.source.TrackGroup;
+import com.google.android.exoplayer2.source.TrackGroupArray;
+import com.google.android.exoplayer2.trackselection.DefaultTrackSelector;
+import com.google.android.exoplayer2.trackselection.MappingTrackSelector;
+
+import org.schabi.newpipe.R;
+
+/**
+ * A listener class to allow setting a specified quality for livestreams when clicking an item on
+ * the quality selector popup menu in video players.
+ */
+public class TrackSelectionListener implements PopupMenu.OnMenuItemClickListener {
+    private final Context context;
+    private final CustomTrackSelector trackSelector;
+    private final int rendererIndex;
+    private final TextView qualityTextView;
+    private final TrackGroupArray rendererTrackGroups;
+
+    /**
+     * Create a {@link TrackSelectionListener}.
+     *
+     * @param context         the player context
+     * @param trackSelector   the {@link CustomTrackSelector} used by the player, which couldn't be
+     *                        null
+     * @param rendererIndex   the index of the video renderer
+     * @param qualityTextView the {@link TextView quality text view} on which the playing quality
+     *                        will be updated after a quality item is clicked
+     * @throws NullPointerException if trackSelector is null
+     */
+    public TrackSelectionListener(@NonNull final Context context,
+                                  @NonNull final CustomTrackSelector trackSelector,
+                                  final int rendererIndex,
+                                  @NonNull final TextView qualityTextView) {
+        this.context = context;
+        this.trackSelector = trackSelector;
+        final MappingTrackSelector.MappedTrackInfo mappedTrackInfo = checkNotNull(
+                trackSelector.getCurrentMappedTrackInfo());
+        this.rendererIndex = rendererIndex;
+        this.qualityTextView = qualityTextView;
+
+        rendererTrackGroups = mappedTrackInfo.getTrackGroups(rendererIndex);
+    }
+
+    @Override
+    public boolean onMenuItemClick(@NonNull final MenuItem item) {
+        final DefaultTrackSelector.Parameters selectionParameters = trackSelector.getParameters();
+        final boolean isDisabled = selectionParameters.getRendererDisabled(rendererIndex);
+        final DefaultTrackSelector.ParametersBuilder builder = selectionParameters.buildUpon()
+                .clearSelectionOverrides(rendererIndex)
+                .setRendererDisabled(rendererIndex, isDisabled);
+        // The auto string
+        final String qualityTextViewText = qualityTextView.getText().toString();
+
+        if (item.getItemId() != 0) {
```

Comment:
Invert the two branches of this `if`, and explain that `0` is always the index of `auto`